dix/rpcbuf.c

```diff
@@ -8,15 +8,29 @@
 
 #include "dix/rpcbuf_priv.h"
 
-Bool x_rpcbuf_makeroom(struct x_rpcbuf *rpcbuf, size_t needed)
+static inline Bool __x_rpcbuf_write_bin_pad(
+    x_rpcbuf_t *rpcbuf, const char *val, size_t len)
+{
+    const size_t blen = pad_to_int32(len);
+
+    char *reserved = x_rpcbuf_reserve(rpcbuf, blen);
+    if (!reserved)
+        return FALSE;
+
+    memcpy(reserved, val, len);
+    memset(reserved + len, 0, blen - len);
+    return TRUE;
+}
+
+Bool x_rpcbuf_makeroom(x_rpcbuf_t *rpcbuf, size_t needed)
 {
     /* break out of alreay in error state */
     if (rpcbuf->error)
         return FALSE;
 
     /* not allocated yet ? */
     if (!rpcbuf->buffer) {
```

Comment:
Might as well remove this whole if-statement (or in another commit), it's not needed as the realloc below will do the same job. Having this here also leads to situations where it can do malloc twice unnecessarily if you request a large amount of bytes as the first allocation.

Reply:
which if statement ? github seems to show the wrong lines here :(

Reply:
The `if (!rpcbuf->buffer) {` codeblock that does malloc. Realloc does the same job below (and it works with a nullpointer)

Reply:
changed it to realloc

Reply:
It looks good now